src/admin/UserList.tsx

```diff
@@ -1,27 +1,77 @@
-import React, { useMemo, useReducer, useRef, useState } from "react";
+import React, { useRef, useState, useEffect } from "react";
 import { getApiUrl } from "../api/api";
 import { Modal, Button } from "react-bootstrap";
 
 import { User } from "../shared/types";
-import ProfileForm from "../profile/ProfileForm";
+import ProfileForm, { useProfileForm } from "../profile/ProfileForm";
 
 import "./UserList.scss";
 import { useAuth } from "../auth/AuthProvider";
 import ERRORS from "../shared/errors";
-import { RichTable, Column, RichTableBag } from "../shared/components/RichTable";
+import {
+  RichTable,
+  Column,
+  RichTableBag,
+} from "../shared/components/RichTable";
 
-interface State {
-  allUsers: User[];
-  filteredUsers: User[] | undefined;
-  currentUser: User | undefined;
-  showEditUser: boolean;
-  showCreateUser: boolean;
+interface UserEditModalProps {
+  user?: User;
+  show: boolean;
+  setShow: (show: boolean) => void;
+  refreshTable: () => void;
 }
 
-type Action =
-  { type: 'set edit user', data: User | undefined } |
-  { type: 'show edit user', data: boolean } |
-  { type: 'show create user', data: boolean };
+const UserEditModal: React.FC<UserEditModalProps> = (
```

Comment:
nit: Can this be moved after columns?

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/22b3f1fd434e243f91d53ffd7c901162d0c89b3b